core/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/HighlightBuilder.java

```diff
@@ -522,4 +525,33 @@ public String toString() {
             return name().toLowerCase(Locale.ROOT);
         }
     }
+
+    public static enum BoundaryScannerType implements Writeable {
+        SIMPLE, BREAK_ITERATOR;
```

Comment:
This is actually exactly what it means. As I wrote in the PR comment, I wanted to get initial feedback. Now that I see you're OK with the general approach, I want to add a "break_iterator_type" property which defaults to "SENTENCE" but also supports "WORD", "CHARACTER" and "LINE".